### DISCUSSION TITLE
ufw action to kill established connections on ban

### DISCUSSION THREAD

**Discussion by Brewer7** [fail2ban/fail2ban#3401] [Q&A] [answered]
Hi everyone,

im using F2B with ufw on Ubuntu 22.04. It generally works apart on 2 issues

1. when the ban rule is added established connections remains. which actually reduce the level of protection

i've created custom action looks like that:
```
[Definition]
actionstart =

actionstop =

actioncheck =

actionban = ufw insert 1 deny from <ip> to any port <port> && conntrack -D -s <ip>

actionunban = ufw delete deny from <ip> to any port <port>
```

**&& conntrack -D -s <ip>** - this part suppose to terminate connections on ban but it seens to be wrong expression

**fail2ban.log:** 
`fail2ban.utils[48510]: ERROR 7fdaa4550150 -- stderr: '/bin/sh: 1: Syntax error: "&&" unexpected'`

How should i ammend the action to kill established connections? 

2. When ban rule added to UFW it's added in a following way
```

#ufw status

Status: active

To                         Action      From
--                         ------      ----
Anywhere                   REJECT      212.xxx.xxx.xxx
```
**jail.local**
```
[roundcube-new]

enabled  = true
filter   = roundcube-auth-new
port = 80,443,25,587,465,110,995,143,993,4190
action = ufw
logpath  = /var/log/mail.log
```

Can i block access on specific port only but not all connections?

Thanks for your advises in advance

**Comment by sebres** [comment, marked as answer]
> How should i ammend the action to kill established connections?

newest ufw action support kill-mode `conntrack` since bbfff1828061514e48395a5dbc5c1f9f81625e82
```ini
banaction = ufw[kill-mode=conntrack]
```

> Syntax error: "&&" unexpected'

It looks like something is wrong with your command, it seems not to have something before `&&` or just a new-line or a semi-colon.
`Syntax error: "&&" unexpected` is only occurring if some command starts with `&&`
```bash
$ sh -c 'echo 1st && echo 2nd'
1st
2nd
$ sh -c 'echo 1st; && echo 2nd'
sh: 1: Syntax error: "&&" unexpected
$ sh -c '&& echo 2nd'
sh: 1: Syntax error: "&&" unexpected
```
(in error case Fail2ban'd provide the whole command string it executes, see the fail2ban.log above the error with same ID `7fdaa4550150`)

> Can i block access on specific port only but not all connections?

With ufw? Not really IIRC, there is something like application, but I have no idea whether it restrict the ports specified for app internally.
I could only suggest - **don't use ufw!** Use native net-filter like iptables/ipset or nftables instead.
